### ISSUE TITLE
Same name tags are created

### ISSUE THREAD

**Issue by Transportman** [firefly-iii/firefly-iii#2030]
**Bug description**
I am running Firefly III version 4.7.9

When selecting a tag after typing the first letters with different capitals than the actual tag, the tag is created again.

**Steps to reproduce**
- Have a tag Example
- Create a new transaction
- In the tag field, type EX and then select the suggested Example tag
- Save the transaction
- There are now 2 Example tags, and only one has the transaction

**Expected behavior**
I expected to only have 1 Example tag

**Extra info**
Firefly and DB is running using the default container images from the repo in a minikube VM on a Windows host.

Debug information generated at 2019-01-27 10:15:03 UTC for Firefly III version **4.7.9**.

| Variable | Content |
| --- | --- |
| FF version | 4.7.9 |
| FF API version | 0.9.0 |
| App environment | local |
| App debug mode | '' |
| App cache driver | file |
| App logging | , stdout |
| PHP version | 7.2.13 |
| Display errors | Off |
| Session start | 2019-01-01 00:00:00 |
| Session end | 2019-01-31 23:59:59 |
| Session first | 2018-12-31 00:00:00 |
| Error reporting | ALL errors |
| Host | Linux |
| Interface | apache2handler |
| UserID | 1 |
| Attempt at "en" | false |
| Attempt at "English" | false |
| Attempt at "en_US.utf8" | 'en_US.utf8' |
| Attempt at "en_US.UTF-8" | 'en_US.UTF-8' |
| DB drivers | mysql, pgsql, sqlite |
| Current driver | pgsql |
| Login provider |   |
| Storage disks | local-upload  |
| Using Sandstorm? | no |
| Is Sandstorm (.env) | false |
| Is Docker (.env) | true |
| bunq uses sandbox | false |
| Trusted proxies (.env) |  |
| User agent | Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:64.0) Gecko/20100101 Firefox/64.0 |
| Loaded extensions | Core, date, libxml, openssl, pcre, sqlite3, zlib, ctype, curl, dom, fileinfo, filter, ftp, hash, iconv, json, mbstring, SPL, PDO, bz2, posix, Reflection, session, SimpleXML, pdo_sqlite, standard, tokenizer, xml, xmlreader, xmlwriter, mysqlnd, apache2handler, bcmath, Phar, gd, intl, ldap, pdo_mysql, pdo_pgsql, sodium, tidy, zip |

**Bonus points**
Earn bonus points by: n/a

- Post a stacktrace from your log files
- Add a screenshot

**Comment by Transportman** [comment]
Strange, for me the reproduction is also not working as I assumed it would be for all tags, as I do observe it for one of my tags, but after further testing, not for other tags.

Maybe you can see something from the database records, I added the two records from the tag-table that I have a problem with (assuming that the ID that is shown in the browser matches the ID in the database). ID 21 is the tag that should be used, named Vast, ID 33 is the additionally created one when I start with VA in the tag field and then select Vast. 

 id |     created_at      |     updated_at      | deleted_at | user_id |
                                               tag
                          | tagMode | date |
               description                                                                                            |
latitude | longitude | zoomLevel
----+---------------------+---------------------+------------+---------+------------------------------------------------
------------------------------------------------------------------------------------------------------------------------
--------------------------+---------+------+----------------------------------------------------------------------------
----------------------------------------------------------------------------------------------------------------------+-
---------+-----------+-----------
 21 | 2019-01-20 10:19:41 | 2019-01-20 10:19:41 |            |       1 | eyJpdiI6IndaT1lkNlZSVmswTUl5TXRNYlZ2UHc9PSIsInZ
hbHVlIjoiQm1BXC9OQktnWitLbkNcLytTTXFaNEdBPT0iLCJtYWMiOiI0ZGVhMTI3OTcxOGFkM2UyMjIwN2ExMzc5NjAwZGQ2ZDBjMTc2OGQwNmZiZWM0NGY
wOTQ1MGY4M2Q3ODRjZjIxIn0= | nothing |      | eyJpdiI6ImswZTduNW50V1NrU2pGV2tuMXZTZmc9PSIsInZhbHVlIjoiRHVEWHA0THE0cHRxTVV
KR2xtR3I5UT09IiwibWFjIjoiMTI1MjcwNzVmYmMyNjk2Y2I4N2YwNzYwNDMzZWRlMjQ1YWY3MGRhOGU0Y2I4ZDY2ODg4MzcyODIyZDBmZDU0OCJ9     |
         |           |
 33 | 2019-01-27 11:03:28 | 2019-01-27 11:03:28 |            |       1 | eyJpdiI6Ino2TStUd1lRWHhQT2NUd3R0TjczT3c9PSIsInZ
hbHVlIjoiZ2tReE45YWQ1cUJQRmpoWXlaeG5cL0E9PSIsIm1hYyI6IjkwMTg1ZDU5NTM0YWMyMmZlMjE1NzQ5YjNmMGYzM2YxODA2NGQ5NWI5NWQ2YzhlYmF
hYTI4ZmNmZjRmYmZjODUifQ== | nothing |      | eyJpdiI6Ik1QWlpPeHprUGJBXC9ndUJrSDN4ZkJ3PT0iLCJ2YWx1ZSI6ImQ2OGloTGJPalN4V0V
lUkpkUkdGcVE9PSIsIm1hYyI6ImM5MjY5YWQ4MTBiNzIwOWUwZTE1MTdiMTU0M2JhMjc2NDEwMTc1MzU5ZmJjMTYyMjQ4Mjg4MTYxMWQzNzllMzQifQ== |
         |           |
(2 rows)

**Comment by JC5** [comment]
The ID is indeed the same, but the content is encrypted. So I can't really tell from these examples. Are the tags literally "Example" or something else?

**Comment by Transportman** [comment]
They both are literally "Vast" as shown in the attached image.
![firefly-bug](https://user-images.githubusercontent.com/47077527/51800578-726d0500-2231-11e9-9726-e4753d02baa8.png)

When I tried to update the one with ID 21, it says that there is already a tag with that name, but the one with ID 33 I can update just fine. Also, creating a new transaction the same way I described, it suggests both tags to me, but does not create an additional tag and adds it to the existing tag with ID 33. 

I now deleted ID 33, renamed the one with ID 21 (just blanked the field and retyped it) and now it no longer creates an additional tag. So it appears to me that the displayed name and the link name during transaction creation were different.

Anyway, since the issue is fixed at my end and it might be some weird edge case, might not be worth spending too much time on it.
